examples/plot_gp_function.py

```diff
@@ -0,0 +1,273 @@
+    
+    fig.suptitle(
+        'Convergences to the optimum after {} iterations'.format(it),
+        fontdict={'size':50}
+    )
+    
+    
+    steps = len(list(optimizers.values())[0].space)
+    n = len(optimizers)
+    colmap = plt.cm.get_cmap(name='rainbow', lut=n)
+    
+    num_iter=np.arange(1,it+1)
+    
+    i=0
+    for acq in optimizers.keys():
+        af.plot(num_iter, points[acq], '*',markersize=15,markerfacecolor=colmap(i), markeredgecolor='k', markeredgewidth=1,label=acq,linestyle='solid',color=colmap(i))
+        i=i+1
+    
+ 
+    af.axhline(y=x[np.argmax(target(x))], linestyle=':', label='Optimum to be achieved')
+    af.legend(loc=2, bbox_to_anchor=(1.01, 1), borderaxespad=0.)
+   
+    
+    # Integer on the x axes
+    a = range(0,21)
+    af.set_xticks(a)
+    af.set_yticks(range(-2,11))
+    plt.xlabel('Number of iterations')
+    plt.ylabel('Suggested x')
+    
+    
+    plt.show()
+
+
+# Plot regret
+
+def simple_regret(f_max, optimizer):
+    """
+    The simple regret rT = max{x∈X} f(x) − max{t∈[1,T]} f(x_t) measures the value of
+the best queried point so far. 
+    """
+    return f_max-optimizer._space.target.max()
+
+def plot_simple_regret(optimizers, x, target, params, dim, it=20):
+    
+    if dim > 1:
+        inputs = []
+        for col in range(dim):
+            inputs.append(x[:,col])
+        y = target(*inputs)
+
+
+        f_max=max(y)
+
+        tar={}
+        points={}
+        utility_function={}
+        regrets={}
+
+
+        for acq, optimizer in optimizers.items():  # acq è la stringa-chiave, optimizer è l'oggetto
+            regrets[acq]=np.zeros(it)
+            utility_function[acq] = UtilityFunction(kind=acq, kappa = params[acq]['kappa'], xi = params[acq]['xi'])
+
+        for i in range(it):
+            for acq, optimizer in optimizers.items():
+                points[acq]=optimizer.suggest(utility_function[acq])
+                tar[acq]= target(**points[acq])
+                optimizer.register(params=points[acq], target=tar[acq])
+                regrets[acq][i]= simple_regret(f_max, optimizer)
+    else:
+        x = x.reshape(-1,1)
+        y = target(x)
+        
+        f_max=max(y)
+
+        tar={}
+        points={}
+        utility_function={}
+        regrets={}
+
+        for acq, optimizer in optimizers.items():  # acq è la stringa-chiave, optimizer è l'oggetto
+            points[acq]=np.zeros(it)
+            tar[acq]= np.zeros(it)
+            regrets[acq]=np.zeros(it)
+            utility_function[acq] = UtilityFunction(kind=acq, kappa = params[acq]['kappa'], xi = params[acq]['xi'])
+
+        for i in range(it):
+            for acq, optimizer in optimizers.items():
+                points[acq][i]=optimizer.suggest(utility_function[acq])['x']
+                tar[acq][i]= target(points[acq][i])
+                optimizer.register(params=points[acq][i], target=tar[acq][i])
+                regrets[acq][i]= simple_regret(f_max, optimizer)
+
```

Comment:
Qui abbiamo dovuto distinguere il caso uni e multi dimensionale perchè dopo molte prove non riuscivamo ad unirli e abbiamo visto che anche l'autore della libreria in "advance_tour" e "visualization" distingue i due casi e ci lavora in modo diverso.